core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -316,6 +347,47 @@ trait EntityComponent {
           concatSqlActions(filteredQuery, filterSql("and", "e")).as[Int]
         }
 
+        // the full query to generate the page of results, as requested by the user
+        def pageQuery = {
+          // did the user request any attributes other than "name" and "entityType"?
+          val isRequestingAttrs: Boolean = entityQuery.fields.fields.forall { fieldSel =>
+            (fieldSel diff Set("name", "entityType")).nonEmpty
+          }
+
+          // different cases for which query to execute:
+          if (entityQuery.sortField == "name" && !isRequestingAttrs) {
+            // simplest; the user is sorting by ENTITY.name and doesn't want any attributes; we only need to
+            // query the ENTITY table
+            // NB: the "null" as the last column in the select is important! It allows this query to be translated into
+            // EntityAndAttributesResult objects; the null represents the lack of any attributes for this entity
+            concatSqlActions(
+              sql"""select e.id, e.name, e.entity_type, e.workspace_id, e.record_version, e.deleted, e.deleted_date, null
+                             from ENTITY e
+                             where e.deleted = 'false' and e.entity_type = $entityType and e.workspace_id = ${workspaceContext.workspaceIdAsUUID} """,
+              filterSql("and", "e"),
+              order("e"),
+              sql" limit #${entityQuery.pageSize} offset #${(entityQuery.page-1) * entityQuery.pageSize}").as[EntityAndAttributesResult]
+          } else {
+            // user is sorting by an attribute value, so we need the largest number of joins
+            // this query is very similar to baseEntityAndAttributeSql
+            /* TODO: include "where e.deleted = 'false' and e.entity_type = $entityType and e.workspace_id = $workspaceId"
```

Comment:
Don't we need the clause to filter out the deleted entities? I've walked through the code but perhaps I'm missing something

Reply:
in this branch of the code - the `else` clause at line 370 - the `e.deleted = 'false'` is included inside `paginationSubquery`, which is called from `paginationJoin`. So, it's functionally correct and returns only active entities. What I'm suggesting in this comment is to *also* include `e.deleted = 'false'` in the first select clause - at lines 380-384 - as an optimization for MySQL. It's possible that by filtering that first select clause, MySQL will have less internal work to do inside its query planner because there's less data. But it also may not help the planner, and might just be extra SQL; it needs some investigation.